src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java

```diff
@@ -160,7 +174,95 @@ public void rateLimitsBlockBuilds(int rate) throws Exception {
             assertThat("At least the rate implied delay but no more than 500ms longer",
                     System.currentTimeMillis() - startTime,
                     allOf(
-                            greaterThanOrEqualTo(60 * 60 / rate * 1000L),
+                            greaterThanOrEqualTo(60 * 60 / rate * 1000L - 200L),
+                            lessThanOrEqualTo(60 * 60 / rate * 1000L * 500L)
+                    )
+            );
+            future.get();
+        }
+
+    }
+
+    @Test
+    public void rateLimitsConcurrentBuilds() throws Exception {
```

Comment:
only because I removed the event based test that found the bug about concurrent entries in the queue